src/core/processing/processingalgorithm.cpp

```diff
@@ -268,8 +268,58 @@ bool ProcessingAlgorithm::run( bool previewMode )
     }
     else
     {
-      // Currently, only feature-based algorithms are supported
-      return false;
+      parameters[QStringLiteral( "INPUT" )] = QgsProcessingFeatureSourceDefinition( mInPlaceLayer->id(),
+                                                                                    false,
+                                                                                    -1,
+                                                                                    Qgis::ProcessingFeatureSourceDefinitionFlags(),
+                                                                                    Qgis::InvalidGeometryCheck(),
+                                                                                    QStringLiteral( "$id IN (%1)" ).arg( featureIds.join( ',' ) ) );
+      parameters[QStringLiteral( "OUTPUT" )] = QStringLiteral( "memory:" );
+
+      bool ok;
+      QVariantMap results = mAlgorithm->run( parameters, context, &feedback, &ok, { { QStringLiteral( "IN_PLACE" ), true } } );
+      if ( ok )
```

Comment:
What in case is not `ok`? Should we show the error somehow?

Reply:
Error handling needs thinking in separate PR for both feature based and this type of alg.